toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt

```diff
@@ -275,6 +283,19 @@ class CratesLocalIndexServiceImpl
 
         private val LOG: Logger = logger<CratesLocalIndexServiceImpl>()
     }
+
+    @TestOnly
+    var indexedCommitHash: String = INVALID_COMMIT_HASH
+
+    /**
+     * Simulates the behaviour of real service.
+     * Loads crates local index in current thread and updates local [indexedCommitHash].
+     */
+    @TestOnly
+    fun updateCrates() {
+        updateCrates(registryHeadCommitHash, indexedCommitHash)
+        indexedCommitHash = registryHeadCommitHash
+    }
```

Comment:
Why do we need this additional code? Let's better use the existing one. I.e. we already have a place where the service keeps commit hash. Also, we already have a method to update info about crate registry - `updateIfNeeded`. Note, although `updateIfNeeded` uses background task to do it, in tests `com.intellij.openapi.progress.Task#queue` is synchronous so let's better use it instead.
Otherwise, we use a lot of code that is never called in production. That makes our tests less useful than they can be

Reply:
> in tests `com.intellij.openapi.progress.Task#queue` is synchronous

Wow, I didn't know about it either, thanks for the clarification!